src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -308,6 +314,30 @@ private async Task<List<DeploymentMachine>> GetDeploymentTargetsAsync(AgentSetti
 
             return machines;
         }
+
+        private string GetAzureSubscriptionId()
+        {
+            string azureSubscriptionId = string.Empty;
+            const string imdsUri = "http://169.254.169.254/metadata/instance/compute/subscriptionId?api-version=2017-08-01&format=text";
```

Comment:
what is `http://169.254.169.254`? seem odd we hard code a http IP in  source code.

Reply:
Azure Instance Metadata Service is a REST endpoint available to all Azure VM's. This endpoint is a non-routable well known address ( 169.254.169.254 ). We are using this in order to fetch metadata about the subscription used to provision that VM. [More here](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service)